src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -58,13 +66,20 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
                     }
                 }
 
-                Console.CancelKeyPress += Console_CancelKeyPress;
+                if (mode == ExtensionModes.Execute)
```

Comment:
Rather than making this method serve a dual purpose, let's split the validation into a separate method.

I did a decent refactor in #4247. If we take those changes, you can easily create the service collection, pull out the `IOptionsSnapshot<T>` from it, and call `Get`. That will then throw an `OptionsValidationException` that you can report the results.